src/sentry/api/base.py

```diff
@@ -450,6 +454,20 @@ def dispatch(self, request: Request, *args, **kwargs) -> Response:
                     span.set_data("SENTRY_API_RESPONSE_DELAY", settings.SENTRY_API_RESPONSE_DELAY)
                     time.sleep(settings.SENTRY_API_RESPONSE_DELAY / 1000.0 - duration)
 
+        # Only enforced in dev environment
+        if settings.ENFORCE_PAGINATION:
+            if request.method.lower() == "get":
+                # Response can either be Response or HttpResponse, check if it's value an array
+                if hasattr(self.response, "data") and isinstance(self.response.data, list):
+                    # if not paginated and not in  settings.SENTRY_API_PAGINATION_ALLOWLIST, raise error
+                    if (
+                        handler.__self__.__class__.__name__
+                        not in settings.SENTRY_API_PAGINATION_ALLOWLIST
+                        and not self.has_pagination(self.response)
+                    ):
+                        raise MissingPaginationError(
+                            f"Response is not paginated correctly in {handler.__func__.__qualname__}, {self.response.headers}, {self.response.data}"
```

Comment:
Let's add more context to this error since it will only be internally visible to sentry employees. Something like:
```
List API response should be paginated otherwise it can break the product in future caused by eventual growth. Learn more about pagination in https://docs.sentry.io/api/pagination/ and reach out to #discuss-api if you have any questions.
```